include/uxr/agent/middleware/Middleware.hpp

```diff
@@ -75,23 +73,25 @@ class Middleware
 
     /* Write and read functions. */
     virtual bool write_data(uint16_t datawriter_id, std::vector<uint8_t>& data) = 0;
+    virtual bool set_read_cb(uint16_t datareader_id, OnNewData on_new_data_cb) = 0;
+    virtual bool unset_read_cb(uint16_t datareader_id) = 0;
     virtual bool read_data(uint16_t datareader_id, std::vector<uint8_t>* data) = 0;
 
     /* Matching functions. */
-    virtual bool matched_participant_from_ref(uint16_t participant_id, const std::string& ref) = 0;
-    virtual bool matched_participant_from_xml(uint16_t participant_id, const std::string& xml) = 0;
+    virtual bool matched_participant_from_ref(uint16_t participant_id, const std::string& ref) const = 0;
+    virtual bool matched_participant_from_xml(uint16_t participant_id, const std::string& xml) const = 0;
 
-    virtual bool matched_topic_from_ref(uint16_t topic_id, const std::string& ref) = 0;
-    virtual bool matched_topic_from_xml(uint16_t topic_id, const std::string& xml) = 0;
+    virtual bool matched_topic_from_ref(uint16_t topic_id, const std::string& ref) const = 0;
+    virtual bool matched_topic_from_xml(uint16_t topic_id, const std::string& xml) const = 0;
 
-    virtual bool matched_datawriter_from_ref(uint16_t datawriter_id, const std::string& ref) = 0;
-    virtual bool matched_datawriter_from_xml(uint16_t datawriter_id, const std::string& xml) = 0;
+    virtual bool matched_datawriter_from_ref(uint16_t datawriter_id, const std::string& ref) const = 0;
+    virtual bool matched_datawriter_from_xml(uint16_t datawriter_id, const std::string& xml) const = 0;
 
-    virtual bool matched_datareader_from_ref(uint16_t datawriter_id, const std::string& ref) = 0;
-    virtual bool matched_datareader_from_xml(uint16_t datawriter_id, const std::string& xml) = 0;
+    virtual bool matched_datareader_from_ref(uint16_t datawriter_id, const std::string& ref) const = 0;
```

Comment:
The variable name `datawriter_id` must be `datareader_id`

Reply:
Fixed!